src/workloads/docs/HelloWorld.yml

```diff
@@ -33,6 +33,13 @@ Actors:
       ^NumExpr:
         withExpression: "messages / threads"
         andValues: {messages: *TotalMessageCount, threads: *Threads}
+  # To concatenate strings, use the ^FormatString preprocessor directive. It works similarly to the
+  # ^FormatString generator.
+  - Message:
+      ^FormatString:
```

Comment:
How do we distinguish the pre-processor version of this and the generator version of this? I'm worried that we are going to process all of them.

Reply:
Well, it _does_ process all of them, but any that fail will be left as a `^FormatString` (although the inner docs are still recursively processed). So basically, if it fails to do the formatting at preprocess time, it'll attempt to run it as a generator.

Happy to change it up to something like `^PreprocessorFormatString`, but my concern was primarily that this requires genny users to be fairly aware of what is preprocessed vs generator-able.

Reply:
After looking and thinking a while, I think I would like a different name for the preprocessed version. I think we should be explicit about what is continually regenerated at runtime versus what is evaluated once before running. 

I haven't been able to come up with any truly bad failure cases, but I could still imagine someone wondering why they can't use a  ^RandomString inside a ^FormatString when defining the actor name, but they could when creating a document within their workload. 

@rtimmons thoughts?

Reply:
I think that's valid. I couldn't fully decide on which behavior I wanted, so either is fine by me.

Reply:
If we lean into lisp-isms, preprocessing is a macro. Most lisps have a thing that calls out macros in the syntax. I like `^PreprocessorFormatString` although I wonder if a better syntax like `^^` for preprocessor things would be more intuitive long-term.

Reply:
Created PERF-4543 to track this. Also realized that this probably belongs in the EVG queue...